giphy-search/src/components/GifContainer-solution.jsx

```diff
@@ -0,0 +1,45 @@
+/* 
+This component is meant to display gifs. However, there are two different sets of gifs that this component can display, depending on the user's actions. At first, they will see trending gifs but after submitting a search term in the GifSearch form, they will see gifs related to their search.
+
+useState — we create a piece of state for the fetched gifs. Depending on the user's actions, this state will either be trending gifs or searched gifs.
+
+useEffect — we run the effect to fetch the gifs using the getTrendingGifs adapter and set the gifs state. If the searchTerm prop changes, we will run the effect  again, this time using the getGifsBySearch adapter.
+*/
+import defaultGifs from '../gifs.json'
+import { useEffect, useState } from 'react';
+import { getGifsBySearch, getTrendingGifs } from '../adapters/giphyAdapters-solution';
+
+const GifContainer = ({ searchTerm }) => {
+    const [gifs, setGifs] = useState(defaultGifs);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        const fetchGifs = async () => {
+            const [data, error] = searchTerm ? await getGifsBySearch(searchTerm) : await getTrendingGifs();
+            if (error) {
+                console.log('error')
+                setError(error);
+                setGifs(defaultGifs);
+            }
+            else if (data) setGifs(data);
+        }
+        fetchGifs();
+    }, [searchTerm]);
+
+    return (
+        <>
+            {error ? (<p>Sorry, the GIPHY API is not working, but here are some cats coding...</p>) : ""}
+            <ul>
```

Comment:
I've added in some proper error handling here as well as a bonus. It's good to bring awareness to error handling, even if not strictly required. It will push the more advanced students.